drainer/sync/kafka.go

```diff
@@ -106,85 +143,239 @@ func NewKafka(cfg *DBConfig, tableInfoGetter translator.TableInfoGetter) (*Kafka
 	return executor, nil
 }
 
+func (ks *KafkaSyncer) getPartitions() ([]int32, error) {
+	return ks.cli.Partitions(ks.topic)
+}
+
 // Sync implements Syncer interface
-func (p *KafkaSyncer) Sync(item *Item) error {
-	slaveBinlog, err := translator.TiBinlogToSlaveBinlog(p.tableInfoGetter, item.Schema, item.Table, item.Binlog, item.PrewriteValue)
+func (ks *KafkaSyncer) Sync(item *Item) error {
+	slaveBinlog, err := translator.TiBinlogToSlaveBinlog(ks.tableInfoGetter, item.Schema, item.Table, item.Binlog, item.PrewriteValue)
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if ks.partitionMode == PartitionFixed {
+		return errors.Trace(ks.saveBinlog(slaveBinlog, item))
+	}
 
-	err = p.saveBinlog(slaveBinlog, item)
+	var msgs []*sarama.ProducerMessage
+	switch ks.partitionMode {
+	case PartitionBySchema:
+		msgs, err = ks.splitBinlogBySchema(slaveBinlog, item)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	case PartitionByTable:
+		msgs, err = ks.splitBinlogByTable(slaveBinlog, item)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	default:
+		return errors.Errorf("Not supported partition mode: %v", ks.partitionMode)
+	}
+	if ks.includeResolvedTs {
```

Comment:
don't send resolve to every partition for every txn? limit by some period like send one for every 5s?